ARE/components/processor.serialport/src/main/java/eu/asterics/component/processor/serialport/SerialPortInstance.java

```diff
@@ -377,60 +399,86 @@ public void handlePacketReceived(byte actbyte) {
+    }
 
-        portController = CIMPortManager.getInstance().getRawConnection(propComPort, propBaudRate, true);
-        received = "";
+    private void init() {
 
+        received = "";
+        initComPort();
         if (portController == null) {
             AstericsErrorHandling.instance.reportError(this,
                     "SerialPort-plugin: Could not construct raw port controller, please verify that the COM port is valid.");
-        } else {
-            in = portController.getInputStream();
-            out = portController.getOutputStream();
-            readerThread = new Thread(new Runnable() {
-
-                @Override
-                public void run() {
-                    running = true;
-                    while (running) {
+            return;
+        }
+        in = portController.getInputStream();
+        out = portController.getOutputStream();
+        readerThread = new Thread(new Runnable() {
 
-                        try {
-                            if (in.available() > 0) {
-                                handlePacketReceived((byte) in.read());
-                            } else {
-                                Thread.sleep(10);
-                            }
-                        } catch (InterruptedException ie) {
-                            ie.printStackTrace();
-                        } catch (IOException io) {
-                            io.printStackTrace();
+            @Override
+            public void run() {
+                running = true;
+                try {
+                    while (running) {
+                        Byte read = portController.poll();
+                        if (read != null) {
+                            handlePacketReceived(read);
                         }
-
                     }
+                } catch (IOException e) {
+                    logger.log(Level.WARNING, "IOException in polling data in SerialPort module.");
+                    closeAll();
                 }
+                logger.log(Level.FINE, "SerialPort module: stopped reading thread.");
+            }
+        });
+        readerThread.start();
+        logger.log(Level.FINE, "SerialPort module: started reading thread.");
 
-            });
-            readerThread.start();
+    }
 
+    private void initComPort() {
+        //Sanity check: if portController != null stop the connection first.
+        if (portController != null) {
+            stop();
+        }
+
+        if (propCimId != null) { //mode with given CIM id
+            initComPortByCimID();
+        } else { //mode with given COM Port
+            portController = CIMPortManager.getInstance().getRawConnection(propComPort, propBaudRate, true);
+        }
+    }
+
+    private void initComPortByCimID() {
+        if(CIMPortManager.getInstance().inRescan()) {
+            logger.info("do not try open COM port by CIM id, because currently rescanning");
+            opReceived.sendData(IN_PORT_RESCAN.getBytes());
```

Comment:
So you send the text IN_PORT_RESCAN to the received output port?
I thinks this is not good because then data and status messages are mixed.

Why not adding a new port e.g. 
opPortStatus?
or adding some event triggerer: 
PortScanningStarted
PortScanningStopped
??